### DISCUSSION TITLE
This project is not dead. Because is live in our heart

### DISCUSSION THREAD

**Discussion by lodos2005** [noisetorch/NoiseTorch#266] [General]
@lawl thank you for this awesome project. 

we will always love you and this project. ❤️

**Comment by Senanfurkan** [comment]
You failed no one. Thank you for everything you have done so far.

**Comment by reynoldsme** [comment]
NoiseTorch has been immensely helpful in allowing me to communicate effectively with people since the start of the pandemic and I'm sure the same is true for thousands of other users. I can't thank the maintainer and contributors enough for providing such a useful tool to the community.

If a full audit of the code is needed, I'm sure the community can come together to do what is required. I truly hope this isn't the end of NoiseTorch in its current form.

**Comment by aniketfuryrocks** [comment]
I don't understand what has happened or what is happening. Can't we just use the most recent fork and create a new GitHub organization out of it?

**Comment by contraexemplo** [comment]
@aniketfuryrocks From what I've gathered, one of lawl's systems has been compromised. That system stored their private keys. lawl isn't sure if whoever compromised their system has done anything to this project, so they've advised us to not trust the binaries or the source code.

**Comment by lawl** [comment]
@contraexemplo is correct. Sorry i feel dead right now. I don't have any energy to deal with this right now.

I believe the compromise is unrelated, but again, better safe than sorry.

**Comment by contraexemplo** [comment]
@lawl I'm really sorry you're going through this. Please take all the time you need to recover!
 
One thing to keep in mind, though, is that you've built a community on top of a really useful and easy to use software. We'd love to help you with whatever you may need to keep the project going. You'll have a community to go back to when you're ready.

**Comment by Foexle11** [comment]
I know, I'm just an ordinary User, but if we as community can help in any way, please let us know <3

**Comment by ryleu** [comment]
@lawl Thank you so much for this awesome project, please don't give up on it!

To move forward, however, I need to know the scope of the breach, including how long this has been going on for. What's the nature of the compromise? I need to know if my systems are still secure, or if I need to delete everything.

**Comment by lawl** [comment]
> To move forward, however, I need to know the scope of the breach, including how long this has been going on for. What's the nature of the compromise? I need to know if my systems are still secure, or if I need to delete everything.

I'd like to know that too.

**Comment by ryleu** [comment]
Alright, then how can I help you find out? Are there any commits in particular that are likely suspicious that I should look at? I'm going to start digging through past commits and see if I can find anything.

**Comment by lawl** [comment]
https://github.com/lawl/NoiseTorch/commit/8c34658b64f1efeab501bef57d2bfa9579fe34e2
https://github.com/lawl/NoiseTorch/commit/38787e4195f2a34d7ec4421caf17cb99bc31fa2b

are imo the most likely candidates where one would hide a backdoor, i had reviewed the entire diff i vendored. I take dependencies seriously. But who knows, if my system was compromised, was it showing the right things?

How would one trust *your* review?

edit: least -> most

sorry, still slightly foggy head operating from a system i don't trust.

**Comment by ryleu** [comment]
I'm doing my review from a device that has never touched NoiseTorch. Unless the person who compromised your system is a super genius, with the foresight to edit how others' browsers view the getfedora.org page, *including* the checksums available there, my device is not compromised.

As for how you can trust me personally? You can't. But if enough other people look at the code with me, eventually you should have enough of a croudsourced audit that you can move on safely.

I'll look at those commits and then I'll start digging around in the source. I'm going to work on a testing environment to watch for unexpected behavior from NoiseTorch as well, but that may take some time, because I don't have a lot of experience there.

**Comment by ryleu** [comment]
For speed's sake, I'm going to assume that official GoLang packages are safe to use, and will instead focus on the ones that come from GitHub repositories.

Edit: oh. there is more to the commits than i thought.

**Comment by lawl** [comment]
> For speed's sake, I'm going to assume that official GoLang packages are safe to use

yes, i consider golang.org part of my trusted computing base, since well, i'm using their compiler, so, we have all read the reflections on trusting trust paper, right?

**Comment by ZyanKLee** [comment]
I'm not ready yet to give up on this project:

cadmus has been updated 2 years ago and all other utilities that could fill this gap are either way older (and technologically totally outdated - i.e. pipewire support) or are commercial tools that just would not fit my wallet.

Also noisetorch has been my favourite for quite some time now. UI and UX are good, sound quality on pulseaudio and pipewire are great. 

And finally: while a little dramatic for my taste, the way @lawl handled the incident is great. You apparently informed the world about the breach as soon as you got to know about it yourself and helped the community to check the code as good as you could.

Sure reinstating the trust into this project may take a few hours of checking and auditing, but in the end - I'm sure about it - this project will hopefully rise and shine again.

My suggestion to you, lawl, would be: get a good nights sleep and some rest from this stress. Sort out your machines and make them yours again. We can wait a few weeks or months and the world will still continue turning.

**Comment by ghost** [comment]
Hello! First, thanks to the developer for all the work, and sorry for this incident… I hope you will be able to find some rest after all of this.

Just to know, as a normal user, what should I do besides stop using it for the moment?

**Comment by Technetium1** [comment]
@Rififia That's about all there is to do right now. Keep an eye out for updates.

**Comment by Fuseteam** [comment]
looking forward to see the restoration to be completed soon (or as soon as the README pr gets accepted)
i guess only that and a new github release is needed now :3

**Comment by ryleu** [comment]
I have finished my code review. I had to learn GoLang. As far as I can tell, there is nothing malicious that made it in. I have also been running noisetorch in a testing environment (Ubuntu 18.04 VM) for the duration, with no strange requests captured in wireshark. I would like a second or third opinion, but I am giving this my stamp of approval.

**Reply by ZyanKLee** [reply]
Thanks for taking the effort and your feedback.

I'm happy about every one who checks the code and points out potential optimizations, too. Even when no malicious code was found, this can help a lot.

**Comment by ghost** [comment]
So I assume it is safe to use noisetorch again. Just to know, was the update server verified (i.e., no fake update were triggered? I don't recall anything like this.)? And only the last binary has been checked?

**Reply by ZyanKLee** [reply]
We don't know if the update server has been compromised. To my knowledge, aside from lawl no one in this community has had access to it to verify or check anything.
To avoid such situation in the future, we changed the update process to use GitHub releases instead. So now it's all in the open.

@AXDOOMER unpacked and decompiled several versions of the binary from various points in time (before, around and after the time of the suspected compromise), to see if there was anything fishy in them. They did not find anything.

**Comment by contraexemplo** [comment]
I just wanted to register my support for the community here. I've been following the code audit closely and I'm so happy NoiseTorch is back in business! I tried several different noise suppression solutions in the meantime but none worked as well for me as NoiseTorch. I'll keep following the project to see if there's any useful contribution I can make in the future. Awesome job, everyone!

**Comment by Fuseteam** [comment]
yay~ noisetorch is back~ and it appears to solve my freerdp call audio echo issue~ i suppose this can now be closed: https://github.com/noisetorch/NoiseTorch/issues/274
